src/plugins/data/server/search/es_search/es_search_strategy.ts

```diff
@@ -16,74 +16,59 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { Observable, from } from 'rxjs';
-import { first } from 'rxjs/operators';
-import { SharedGlobalConfig, Logger } from 'kibana/server';
-import { SearchResponse } from 'elasticsearch';
-import { ApiResponse } from '@elastic/elasticsearch';
+import { Logger } from 'kibana/server';
+import { Observable } from 'rxjs';
+import { switchMap, first } from 'rxjs/operators';
+
+import type { SharedGlobalConfig } from 'kibana/server';
+
 import { SearchUsage } from '../collectors/usage';
-import { toSnakeCase } from './to_snake_case';
 import {
-  ISearchStrategy,
-  getDefaultSearchParams,
-  getTotalLoaded,
-  getShardTimeout,
-  shimAbortSignal,
-  IEsSearchResponse,
-} from '..';
+  getSearchArgs,
+  doSearch,
+  includeTotalLoaded,
+  toKibanaSearchResponse,
+} from './es_search_rxjs_helpers';
+import { getShardTimeout } from '..';
+
+import type { ISearchStrategy } from '..';
 
 export const esSearchStrategyProvider = (
   config$: Observable<SharedGlobalConfig>,
   logger: Logger,
   usage?: SearchUsage
-): ISearchStrategy => {
-  return {
-    search: (request, options, context) =>
-      from(
-        new Promise<IEsSearchResponse>(async (resolve, reject) => {
-          logger.debug(`search ${request.params?.index}`);
-          const config = await config$.pipe(first()).toPromise();
-          const uiSettingsClient = await context.core.uiSettings.client;
+): ISearchStrategy => ({
+  search: (request, { abortSignal }, context) => {
+    // Only default index pattern type is supported here.
+    // See data_enhanced for other type support.
+    if (!!request.indexType) {
+      throw new Error(`Unsupported index pattern type ${request.indexType}`);
+    }
 
-          // Only default index pattern type is supported here.
-          // See data_enhanced for other type support.
-          if (!!request.indexType) {
-            throw new Error(`Unsupported index pattern type ${request.indexType}`);
-          }
+    return config$.pipe(
+      getSearchArgs(context.core.uiSettings.client, (defaultParams, config) => {
+        delete defaultParams.ignoreThrottled;
```

Comment:
I understand this isn't part of this PR, but I'd really like to remove `ignoreThrottled` out of OSS code completely (including `getSearchParams`, and introduce a new method that's used only in X-Pack that utilizes `getSearchParams`).